Adding a delete-oldest role to delete old images.

mbruzek opened the pull request:
This PR is to fix #304 
@chaitanyaenr please review

chaitanyaenr commented on image_provisioner/playbooks/delete_oldest.yaml:
```diff
@@ -0,0 +1,5 @@
+---
+- name: Delete the oldest file in the directory
+  hosts: localhost
+  roles:
```
Instead of a separate playbook, can you call this role from existing playbooks?

chaitanyaenr replied:
Also can you make this role optional?

mbruzek replied:
@ravi yes we can do that. I will add it to the provision_gold_images.yaml playbook.

chaitanyaenr commented on image_provisioner/playbooks/roles/delete-oldest/tasks/main.yaml:
```diff
@@ -0,0 +1,74 @@
+- name: Checking for target_directory
+  fail:
+    msg: "Must provide the target_directory variable."
+  when: target_directory is not defined
```
can you add the target_directory variable to the sample inventory files?

mbruzek replied:
I can add the variable to the files, but the way our Jenkins calls it is by copying an inventory outside of this repository so I will also add it to the inventory that Jenkins uses.

sjug commented:
Is deletion of a single image the desired functionality?

mbruzek commented:
This latest commit addresses the review comments from @chaitanyaenr and @sjug. I added the variable to the inventory files and tied it to `provision_gold_images.yaml` rather than having a separate playbook.

@sjug re: desired functionality. We ran out of disk space on the image server (several times) and came up with a solution to delete one image when attempting to create one image to avoid filling up the disk in the future.

chaitanyaenr commented on image_provisioner/playbooks/roles/delete-oldest/tasks/main.yaml:
```diff
@@ -0,0 +1,74 @@
+---
+# Abort if target_directory is not set.
+- name: Check for the target_directory variable
```
I don't think we need this check since the role is run only when the target_directory variable is set.

mbruzek replied:
Since that check is in a different file I would like to keep the check here. If anyone tries to call this role without that variable set, they will know exactly what is wrong and what to fix.

chaitanyaenr commented:
LGTM
